tests/basic/basic_test.go

```diff
@@ -194,6 +194,47 @@ var _ = Describe("AppTasksDown", func() {
 	})
 })
 
+// This test scales up and down an application and checks if app has actually scaled accordingly
+var _ = Describe("AppScaleUpAndDown", func() {
+	It("has to scale up and scale down the app", func() {
+		var contexts []*scheduler.Context
+		for i := 0; i < Inst().ScaleFactor; i++ {
+			contexts = append(contexts, ScheduleAndValidate(fmt.Sprintf("applicationscaleupdown-%d", i))...)
+		}
+
+		Step("scale up all deployments/stateful sets ", func() {
```

Comment:
don't mention deployment or stateful set.. this test should not know constructs from the schedulers.. just say application